### ISSUE TITLE
Hard link support

### ISSUE THREAD

**Issue by kanpov** [AspectUnk/russh-sftp#40]
Title.

A quick google landed me on this:
```
10. sftp: Extension request "hardlink@openssh.com"

This request is for creating a hard link to a regular file. This
request is implemented as a SSH_FXP_EXTENDED request with the
following format:

	uint32		id
	string		"hardlink@openssh.com"
	string		oldpath
	string		newpath

On receiving this request the server will perform the operation
link(oldpath, newpath) and will respond with a SSH_FXP_STATUS message.
This extension is advertised in the SSH_FXP_VERSION hello with version
"1".
```

**Comment by kanpov** [comment]
So, theoretically, this wouldn't be difficult

**Comment by AspectUnk** [comment]
Published as `2.0.2` 🚀

**Comment by kanpov** [comment]
Thanks @AspectUnk !
